clients/fides-js/src/lib/i18n/i18n-utils.ts

```diff
@@ -181,39 +182,45 @@ export function loadMessagesFromExperience(
   experience: Partial<PrivacyExperience>
 ): Locale[] {
   const allMessages: Record<Locale, Messages> = {};
+  let availableLocales: Locale[] = [];
 
   // Extract messages from experience_config.translations
   if (experience?.experience_config) {
+    const config = experience.experience_config;
     const extracted: Record<Locale, Messages> =
-      extractMessagesFromExperienceConfig(experience.experience_config);
+      extractMessagesFromExperienceConfig(config);
     Object.keys(extracted).forEach((locale) => {
       allMessages[locale] = {
         ...extracted[locale],
         ...allMessages[locale],
       };
     });
-  }
-  let availableLocales: Locale[] = Object.keys(allMessages);
 
-  // Extract messages from gvl_translations, filtering to only locales that
-  // exist in experience_config above
-  if (experience?.gvl_translations) {
-    const extracted: Record<Locale, Messages> =
-      extractMessagesFromGVLTranslations(
-        experience.gvl_translations,
-        availableLocales
-      );
-    // Filter the locales further to include only those that existed in
-    // gvl_translations as well
-    availableLocales = Object.keys(extracted);
-
-    // Combine extracted messages with those from experience_config above
-    Object.keys(extracted).forEach((locale) => {
-      allMessages[locale] = {
-        ...extracted[locale],
-        ...allMessages[locale],
-      };
-    });
+    // Set availableLocales to all locales extracted from the experience_config
+    availableLocales = Object.keys(allMessages);
+
+    // Extract messages from gvl_translations, filtering to only availableLocales
+    if (
+      config.component === ComponentType.TCF_OVERLAY &&
+      experience?.gvl_translations
+    ) {
```

Comment:
This `if` on the component is really all that was needed here. I decided to nest this inside `if (experience?.experience_config)` as a side effect